### ISSUE TITLE
Files are not watched correctly

### ISSUE THREAD

**Issue by black-eye** [yongkangchen/remote-sync#373]
Hi, 
I'm facing a strange problem: I started up a new project and established new remote ftp connection. So far so good, changed files are uploaded to ftp.

Than I added a file to watch: /assets/css/style.css. But instead of showing "Currently watching: style.css", "Currently watching: s-power" appeared (see the attachement). It's actually the name of the project, but it's not mentioned there anywhere in this bare form (eg. folders or files names). I'm actually surprised, how he found the name out.

Anyway, the file is not watched, even though the "M icon" sits next to it. I even tried to monitor it manually (thru right click menu), still the same. (The notice was "Currently watching: style.css" this time, but after checking ctrl+shift+P > Remote Sync: Monitor Files List, it was again just "watching: s-power"). 

I tried different combinations of manual and automatic watch settings, Atom restarts etc... None worked.

Strange enough, my old projects (with the similar folder structure) work just fine. It seems that the watcher is watching something else this time.

Any ideas? Can I see somewhere the list of real files being watched?

Thanks a lot
Petr

![obrazek](https://cloud.githubusercontent.com/assets/5634870/23514410/8b212cb4-ff67-11e6-8844-3dc98b1ad666.png)

**Comment by daverickdunn** [comment]
It appears the problem is caused by an exclamation mark in a higher level directory `\!CI2\`. I'm investigating a fix now.

**Comment by enoversum** [comment]
Atom in general does not like special characters in file or folder names. Had the problem that I cannot drag and drop a file from a folder with a space or special character into Atom. This seems to have been fixed, though I ran into problems with special character folder names more often in the past.

**Comment by daverickdunn** [comment]
I don't believe it is an Atom issue. It appears to be an issue with one of the dependencies: [chokidar](https://www.npmjs.com/package/chokidar). This package handles the file monitoring and after some quick investigation appears to be handling exclamation marks in a strange manner. i.e. if a path contains an exclamation mark, it will add several listeners instead of just one.

For example, this path should just add a single file to the chokidar FSWatcher object:

`C:\Users\abc\Desktop\work\spower\!CI2\web\ftpostra2\app\web\css.css`

However, the inclusion of the exclamation mark results in it monitoring:
```
C:\Users\Dave\Desktop\.work
C:\Users\Dave\Desktop\.work\spower
C:\Users\Dave\Desktop\.work\spower\!CI2\web\ftpostra2\jass\app\web\static\css\jass.css
```

I haven't had much time to investigate further, but it's possibly to do with chokidar's globbing functionality, in particular, it's negation feature is suspicious. i.e. pre-pending a path with `!` adds it to an ignore list.

Will hopefully find the exact cause in the coming days.
